src/NetTopologySuite.Lab/Hull/ChiShape.cs

```diff
@@ -0,0 +1,151 @@
+        /// (a) the edge to be removed is longer than the tolerance parameter;
+        /// and (b) the exterior edges of the resulting triangulation form the boundary of a simple polygon;
+        /// (3) Repeat 2. as long as there are more edges to be removed (4) Return the polygon formed by the exterior edges of the triangulation
+        /// </summary>
+        /// <returns>Convex hull if tolerance is higher than the maximum edge length, simple polygon if tolerance is low.</returns>
+        /// <exception>If no concavehull can be generated throws not implemented exception.</exception>
+        public Geometry GetResult()
+        {
+            var subdiv = BuildDelaunay();
+            var tris = ExtractTriangles(subdiv);
+            var hull = ComputeHull(tris);
+            return hull;
+        }
+
+        private static IList<QuadEdgeTriangle> ExtractTriangles(QuadEdgeSubdivision subdiv)
+        {
+            var qeTris = QuadEdgeTriangle.CreateOn(subdiv);
+            return qeTris;
+        }
+
+
+        private Geometry ComputeHull(IList<QuadEdgeTriangle> tris)
+        {
+            var quadEdgeTriangleList = new SortedList<QuadEdge, QuadEdgeTriangle>(new QuadEdgeComparer());
+            foreach (var triangle in tris)
+            {
+                CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, triangle);
+            }
+
+            while (quadEdgeTriangleList.Count != 0)
+            {
+                var triangle = quadEdgeTriangleList.Last();
+                var edge = triangle.Key;
+                Debug.WriteLine(triangle.Value);
+                quadEdgeTriangleList.Remove(edge);
+                if (edge.Length.CompareTo(_tolerance) >= 0 && RemovalGivesRegularPolygon(triangle.Value))
+                {
+                    var neighbours = triangle.Value.GetNeighbours();
+                    triangle.Value.Kill();
+                    foreach (var neighbour in from n in neighbours
+                                              where n != null && n.IsLive()
+                                              select n)
+                    {
+                        Debug.WriteLine(neighbour);
+                        CheckMaxEdgeAndListBorderTriangles(quadEdgeTriangleList, neighbour);
+                    }
+                }
+            }
+            var lineMerger = new LineMerger();
+            var fact = new GeometryFactory();
+            foreach (var triangle in tris)
+            {
+                if (!triangle.IsLive()) continue;
+                for (int i = 0; i < 3; i++)
+                {
+                    if (triangle.GetAdjacentTriangleAcrossEdge(i) == null || !triangle.GetAdjacentTriangleAcrossEdge(i).IsLive())
+                    {
+                        lineMerger.Add(triangle.GetEdge(i).ToLineSegment().ToGeometry(fact));
+                    }
+                }
+            }
+            var mergedLine = (LineString)lineMerger.GetMergedLineStrings().FirstOrDefault();
```

Comment:
nit: these two lines look like they're arguing about whether or not the list is actually allowed to be empty
- if the list **can** be empty, then `if (mergedLine.IsRing)` will throw `NullReferenceException`
- if the list **can't** be empty, then `.FirstOrDefault()` should be more like `[0]`

Reply:
follow-up: "these two lines" was supposed to be a multi-line comment targeting 86 and 87... apparently multi-line comments are still in beta

Reply:
I guess this is another edge case, when the geometry only has one or two points we should not throw something but return a defualt null value.